src/sage/groups/finitely_presented.py

```diff
@@ -1532,7 +1601,169 @@ def alexander_matrix(self, im_gens=None):
+                        A.swap_columns(0, j)
+                        for k in range(1, n):
+                            A.add_multiple_of_row(k, 0, -A[k, 0] * p ** -1)
+                        A = A.delete_rows([0]).delete_columns([0])
+                        n, m = A.dimensions()
+                    else:
+                        if j < m - 1:
+                            j += 1
+                        else:
+                            i += 1
+                            j = 0
+                simpli = unidad
+        return A, R, ideal
+
+    def char_var(self, ring=QQ, matrix_ideal=None, abelianized=None, groebner=False):
+        r"""
+        Return the characteristic varieties of the group ``self``. There are several definitions
+        of the characteristic varieties of a group `G`, see e.g. [CS1999a]_. Let `\Lambda` be the
+        group algebra of `G/G'` and `\mathbb{T}` its associated algebraic variety (a torus). Each
+        element `\xi\in\mathbb{T}` defines a local system of coefficients and the `k` th-characteristic
+        variety is
+
+        .. MATH::
+
+            V_k(G) = \{\xi\in\mathbb{T}\mid \dim H^1(G;\xi)\geq k\}
+
+        These varieties are defined by ideals in `\Lambda`
+
+        INPUT:
+
+        - ``ring`` -- base ring (default: ``QQ``). The base ring of the
+          group algebra
+
+        - ``matrix_ideal`` -- optional. An abelian Alexander matrix and an ideal,
+          to avoid recomputations.
+
+        - ``abelianized`` -- optional. The data of the abelianization.
+
+        - ``groebner`` -- boolean (default: ``False``). If set to
+          ``True`` the minimal associated primes of the ideals and their
+          groebner bases are computed-. It has no effect if the base ring
+          is not a field.
+
+        OUTPUT:
+
+        If ``groebner`` is ``False`` a list of ideals defining the characteristic varieties.
+        If it is True, a list of lists for groebner bases for each ideal.
+
+        EXAMPLES::
+
+            sage: L = [2*(i, j) + 2* (-i, -j) for i, j in ((1, 2), (2, 3), (3, 1))]
+            sage: G=FreeGroup(3) / L
+            sage: G.char_var(groebner=True)
+            [[(f1 - 1, f2 - 1, f3 - 1),
+             (f1 + 1, f2 - 1, f3 - 1),
+             (f1 - 1, f2 - 1, f3 + 1),
+             (f3^2 + 1, f1 - f3, f2 - f3),
+             (f1 - 1, f2 + 1, f3 - 1)],
+             [(f1 - 1, f2 - 1, f3 - 1),
+             (f1*f3 + 1, f2 - 1),
+             (f1*f2 + 1, f3 - 1),
+             (f2*f3 + 1, f1 - 1),
+             (f2*f3 + 1, f1 - f2),
+             (f2*f3 + 1, f1 - f3),
+             (f1*f3 + 1, f2 - f3)]]
+        """
+        if matrix_ideal is None:
+            A, R, ideal = self.abelian_alexander_matrix(ring=ring, abelianized=abelianized, simplified=True)
+        else:
+            A, ideal = matrix_ideal
+            R = A.base_ring()
+        res = []
+        S = R.polynomial_ring()
+        ideal = [S(_) for _ in ideal]
```

Comment:
```suggestion
        ideal = [S(elt) for elt in ideal]
```
The `_` is (by convention) supposed to be used only for iterating but not used, e.g., `[None for _ in range(5)]`.